.github/workflows/build_publish_builder_image.yaml

```diff
@@ -0,0 +1,59 @@
+name: Build and Push Builder Image
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
on push to main are we sure that it is using those default?

Reply:
Ahhh (I forgot), the reason I did that hacky thing was to make sure that push to main triggers the image builds while maintaining control over manual triggers with inputs. And we don't have to do it manually all the time. AFAIK, it doesn't support passing inputs for push triggers. The previous version ensured:
1. For pushes to main, input is empty - rebuild all the images for Go and Ubuntu version metrix
2. For manual triggers, provide an input that you want to build.

Now, either I can revert to old approach or I need to remove push trigger and only keep manual trigger.

Reply:
As discussed, decided to only have push trigger for main. It publishes new images for the recorded go and OS. Will need renovate config to update patch versions for the matrix. Any updates for the matrix will be versioned and should be pushed in order to create and publish image.